libtenzir/builtins/operators/decapsulate.cpp

```diff
@@ -370,7 +371,25 @@ class decapsulate_operator final : public crtp_operator<decapsulate_operator> {
         if (auto diag = parse(row, raw_frame, inferred_type))
           ctrl.diagnostics().emit(std::move(*diag));
       }
-      co_yield builder.finish();
+      // Add back the untouched data column at the end before yielding.
+      auto transformation = indexed_transformation{
+        .index = {slice.columns() - 1},
```

Comment:
Is this the correct index? I tried the sample data and get a segfault:

```
zcat < tenzir/integration/data/pcap/example.pcap.gz | tenzir 'read pcap | decapsulate'
../libtenzir/src/arrow_table_slice.cpp:401: assertion failed 'index out of bounds'
```

Reply:
@dominiklohmann fixed this in 8330237f45.

Please resolve after having taken a look at it.